server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -504,10 +524,27 @@ private void balance() {
                 }
             }
 
-            maybeLogUndesiredAllocationsWarning(allAllocations, undesiredAllocations, routingNodes.size());
+            DesiredBalanceReconciler.this.unassignedShards.set(unassignedShards);
+            DesiredBalanceReconciler.this.undesiredAllocations.set(undesiredAllocations);
+            DesiredBalanceReconciler.this.totalAllocations.set(totalAllocations);
+
+            logReconciliationMetrics.maybeExecute(
+                () -> logger.debug(
+                    new ESLogMessage("DesiredBalanceReconciler stats") //
+                        .field(
+                            "allocator.desired_balance.reconciliation.undesired_allocations",
+                            DesiredBalanceReconciler.this.undesiredAllocations.get()
+                        )
+                        .field(
+                            "allocator.desired_balance.reconciliation.total_allocations",
```

Comment:
Would it make sense to add the percentage as well? because it may be easier to show in a dashboard